snapshots/overlay/overlay.go

```diff
@@ -105,22 +114,44 @@ func NewSnapshotter(root string, opts ...Opt) (snapshots.Snapshotter, error) {
 	if err := os.Mkdir(filepath.Join(root, "snapshots"), 0700); err != nil && !os.IsExist(err) {
 		return nil, err
 	}
-	// figure out whether "userxattr" option is recognized by the kernel && needed
-	userxattr, err := overlayutils.NeedsUserXAttr(root)
-	if err != nil {
-		logrus.WithError(err).Warnf("cannot detect whether \"userxattr\" option needs to be used, assuming to be %v", userxattr)
+
+	if !hasOption(config.mountOptions, "userxattr", false) {
```

Comment:
Looks like you may _silently_ enable `userxattr` even if a user didn't want it, I mean didn't specify it. Is it better to allow a user to choose and if option is specified we check whether kernel supports it, etc or to completely rely on a user's choice without extra checks?

Reply:
That's a breaking change.

Reply:
>No, that's a breaking change.

IIUC, this will be a backward compatibility issue. Anyway, in this case there's no way to "unset" `userxattr` option, even though by default it is "on" this PR adds a possibility to "unset" options.

Another case I see may appear with conflicting options `userxattr|metacopy|redirect_{dir,follow,etc.}`, even though for time being this may be omitted but later this will be the case, because you will need to break this behavior, otherwise it looks like there's no way to support `metacopy|redirect_dir` at least as it is now.

Reply:
In the user namespace it is essential to specify `userxattr`.
https://github.com/torvalds/linux/commit/2d2f2d7322ff43e0fe92bf8cccdc0b09449bf2e1

Outside the user namespace it shouldn't be specified (and actually not specified in this PR).

Reply:
If options conflict with userxattr, a check could be added for it. I would suggest that be done in a follow up though since this change is scoped to adding the options and not any behavior change or new functionality.

Reply:
>Outside the user namespace it shouldn't be specified (and actually not specified in this PR).

As I can see, this PR opens the possibility to do that, e.g. `mount_options=userxattr,...` :D

>If options conflict with userxattr, a check could be added for it. I would suggest that be done in a follow up though since this change is scoped to adding the options and not any behavior change or new functionality

Ok, the last moment from me -- what about to update the documentation (not sure whether it has anything about which options containerd supports and which doesn't) to prevent a user to specify any conflicting options. Also, looks like it is reasonable to at least warn a user that even though this option didn't specify it is enabled.

Reply:
The documentation can be updated in a later PR. We don't always immediately document new features like this, especially as you pointed out, there may be edge cases to figure out that should be pointed out to users.